### PR TITLE
[Data Virtualization Testing] Consume server packages in client that round trip groupId

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#7240](https://dev.azure.com/fluidframework/internal/_workitems/edit/7240)

Bump server packages to allow for `groupId` to be round tripped. We chose a pre-release as there will be more changes made to the local-server as we will want to also control whether or not parts of snapshots are returned via groupId.

### CODE REVIEW

**Review by jatgarg — APPROVED** [review]

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -18068,7 +18068,7 @@ packages:
     engines: {node: ^14.15.0 || ^16.10.0 || >=18.0.0}
     dependencies:
       '@jest/types': 29.6.3
-      '@types/node': 18.19.1
+      '@types/node': 20.10.5
       chalk: 4.1.2
```
Comment (jason-ha) [inline pnpm-lock.yaml:18072]: All of the `@types/node` changes should be reverted.
cc @jatgarg 
Lock files can be annoying to review, but we must.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -19914,7 +19914,7 @@ packages:
       '@previewjs/type-analyzer': 9.0.1
       assert-never: 1.2.1
       prettier: 2.8.8
-      typescript: 5.1.6
+      typescript: 5.3.3
     dev: true
```
Comment (jason-ha) [inline pnpm-lock.yaml:19918]: Another spurious bump that we should roll back.
cc @jatgarg

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -22264,7 +22264,6 @@ packages:
     resolution: {integrity: sha512-nNPsNE65wjMxEKI93yOP+NPGGBJz/PoN3kZsVLee0XMiJolxSekEVD8wRwBUBqkwc7UWop0edW50yrCQW4CyRw==}
     dependencies:
       undici-types: 5.26.5
-    dev: true
```
Comment (jason-ha) [inline pnpm-lock.yaml:22267]: These are the usually easier to spot questionable differences - where something is no longer dev dep.
cc @jatgarg

File: pnpm-lock.yaml (hunk 1 above)
Comment (tyler-cai-microsoft) [reply pnpm-lock.yaml:18072]: https://github.com/microsoft/FluidFramework/pull/19746

File: pnpm-lock.yaml (hunk 3 above)
Comment (tyler-cai-microsoft) [reply pnpm-lock.yaml:22267]: https://github.com/microsoft/FluidFramework/pull/19746

**Pull request merged**